Code/Tools/AssetProcessor/native/ui/MainWindow.h

```diff
@@ -151,8 +155,13 @@ protected Q_SLOTS:
     int m_createJobCount = 0;
     QFileSystemWatcher* m_fileSystemWatcher;
     Config m_config;
+
+    AssetProcessor::BuilderData* m_builderData;
```

Comment:
Can you use smart/shared pointers here for these new pointers?

If not, can you initialize to nullptr here? The existing code might not because some of this was written when we were on an older version of C++ that didn't support initialization in the header.

Reply:
This is the thing that confuses me for a while: how should we reconcile the (AZStd) smart pointers with Qt's pointer management. (I don't like raw pointers either)

I want BuilderData to have a Qt slot so it can update the metrics accordingly. Thus, I made it a QObject.

In MainWindow::Activate, `m_builderData` is created as a Qt child object of MainWindow, and by [Qt document](https://doc.qt.io/qt-6/objecttrees.html) and [StackOverflow answer](https://stackoverflow.com/a/34439533), the child is managed by its parent and is deleted when the parent is deleted. Therefore, I think using raw pointers is better than using AZStd's smart pointer, since if we wrap that in a AZStd smart pointer, the object is deleted twice (once by AZStd smart pointer, once by Qt when MainWindow is destroyed).

How about using (or did you at the first place mean to use) a QPointer? The benefit of it is that when the item is destroyed, the pointer points to `nullptr` instead of an invalid address. If you also think this is the way to go, I will wrap the BuilderData pointer with QPointer around when this object is used, but does not belong to the object that holds this pointer. For here, since the object owns the pointer (MainWindow is BuilderData's parent in Qt), I feel that we should still use a raw pointer and initialize it to `nullptr`.

Reply:
Resolved:
use raw pointers initialized to `nullptr` for objects owned by MainWindow in MainWindow.h
use QPointers in BuilderInfoMetricsModel.h, which uses but does not own the pointed object